source/Halibut.TestUtils.CompatBinary.Base/ProxyServiceForwardingRequestToClient.cs

```diff
@@ -113,5 +110,32 @@ public static async Task Run()
                 }
             }
         }
+
+        public static int WebSocketListeningPort(HalibutRuntime client, out string webSocketListeningUrl)
+        {
+            webSocketListeningUrl = null;
+            for (int i = 0; i < 9; i++)
+            {
+                try
+                {
+                    return WebSocketListeningPortAttemptOnce(client, out webSocketListeningUrl);
+                }
+                catch (HttpListenerException e)
+                {
+                    Console.WriteLine($"Failed to listen for websocket, trying again. {e}");
+                }
+            }
+            return WebSocketListeningPortAttemptOnce(client, out webSocketListeningUrl);
+        }
+        
+        static int WebSocketListeningPortAttemptOnce(HalibutRuntime client, out string webSocketListeningUrl)
```

Comment:
Nit: I'd prefer a dedicated object, or even a tuple rather than an out variable. But this is a low value suggestion 😁